package co.mptc.ecommerce.payment.persistence.mapper;

import co.mptc.ecommerce.payment.domain.entity.CreditEntry;
import co.mptc.ecommerce.payment.persistence.entity.CreditEntryEntity;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface CreditEntryPersistenceMapper {

    // ---------- CreditEntry ----------
    @Mapping(source = "id.value", target = "id")
    @Mapping(source = "customerId.value", target = "customerId")
    @Mapping(source = "totalCreditAmount.amount", target = "totalCreditAmount")
    CreditEntryEntity creditEntryToCreditEntryEntity(CreditEntry creditEntry);

    @Mapping(target = "id.value", source = "id")
    @Mapping(target = "customerId.value", source = "customerId")
    @Mapping(target = "totalCreditAmount.amount", source = "totalCreditAmount")
    CreditEntry creditEntryEntityToCreditEntry(CreditEntryEntity creditEntryEntity);
}
